pkg/operator/gcwatchercontroller/prometheus_client.go

```diff
@@ -62,6 +62,7 @@ func newPrometheusClient(ctx context.Context, configMapClient corev1client.Confi
 	if err != nil {
 		return nil, nil, err
 	}
-
-	return prometheusv1.NewAPI(client), t, nil
+	prometheusClient := prometheusv1.NewAPI(client)
+	_, err = prometheusClient.Rules(ctx)
+	return prometheusClient, t, err
```

Comment:
this would introduce false positives when thanos/prometheus is down for any reason. In that scenario we could not infer the status of garbage collector and the condition would incorrectly be set to false

Reply:
At first, I thought that this is exactly what [this error](https://github.com/openshift/cluster-kube-controller-manager-operator/blob/7f96ac5fb523366079e49e838ae4df83d31709d6/pkg/operator/gcwatchercontroller/gcwatcher_controller.go#L175)  handles, but now I notice that I miss interpret the comment above it.

Reply:
Any  way we can get the GarbageCollectorWatcherController sync method called more often until prometheusClient can actually query the rules?
Otherwise, the kube-controller-manager operator isn't available due to  GarbageCollectorDegraded condition for few minutes.

Reply:
@eranco74 I am trying an alternative approach that should not degrade KCM at all when monitoring stack rollout is in progress: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/706